src/buildercore/project/__init__.py

```diff
@@ -20,49 +22,75 @@ def set_project_alt(pdata, env, altkey):
+    path_list = utils.unique(path_list)
+
+    return path_list
+
+# TODO: consider removing this cache decorator.
+# the expensive parts are in `buildercore.project.files`
 @cache
 def _project_map(project_locations_list=None):
     """returns a single map of all projects and their data"""
-    def merge(orderedDict1, orderedDict2):
-        orderedDict1.update(orderedDict2)
-        return orderedDict1
-
-    project_locations_list = config.app()['project-locations']
-    # ll: {'dummy-project1': {'lax': {'aws': ..., 'vagrant': ..., 'salt': ...}, 'metrics': {...}},
-    #      'dummy-project2': {'example': {}}}
-    data = map(find_project, project_locations_list)
-    opm = reduce(merge, data)
-    # ll: [{'lax': {'aws': ..., 'vagrant': ..., 'salt': ...}, 'metrics': {...}}], {'example': {}}]
-    data = opm.values()
-    # ll: {'lax': {...}, 'metrics': {...}, 'example': {...}}
-
+    def merge(d1, d2):
+        d1.update(d2)
+        return d1
+
+    # a list of paths
+    # ['/path/to/projects.yaml', ...]
+    project_locations_list = parse_path_list(config.PROJECTS_FILES)
+
+    # a list of parsed project data
+    # [{'/path/to/projects.yaml': {'project1': {...}, 'project2': {...}, ...}, {'/path/to/another-projects.yaml': {...}}, ...]
+    data = [files.projects_from_file(path) for path in project_locations_list]
+
+    # a single map of paths to parsed project data
+    # {'/path/to/projects.yaml': {'project1': {...}, 'project2': {...}, ...}, '/path/to/another-projects.yaml': {...}, ...}
+    data = reduce(merge, data)
+
+    # a list of parsed project data.
+    # [{'project1': {...}, 'project2': {...}, ...}, {...}, ...]
+    data = data.values()
+
+    # a single map of parsed project data.
+    # {'project1': {...}, 'project2': {...}, ...}
+    # note: if you have two projects with the same name in different files, one will replace the other.
+    # precedence depends on order of paths in given `project_locations_list`, earlier paths are overridden by later.
     return reduce(merge, data)
 
 def project_map(project_locations_list=None):
-    """returns a deepcopy of the cached `_project_map` results.
-    `cfngen.build_context` is one of probably many functions that are modifying the project data, unintentionally modifying it for all subsequent accesses, including during tests.
-    this approach should be safer, avoid the speed problems with parsing the project files at the cost of a deepcopy."""
+    """returns a single map of all projects and their data.
+    the returned value is a deepcopy of the cached `_project_map` data.
+
+    `cfngen.build_context` is one of probably many functions modifying the project data,
+    unintentionally modifying it for all subsequent accesses including during tests.
+
+    this approach should be safer and avoid the speed problems with parsing the project
+    files again at the cost of a deepcopy."""
     return utils.deepcopy(_project_map(project_locations_list))
```

Comment:
_project_map will need to exclude non-project config, now that stack config lives in the same list